src/charm.py

```diff
@@ -862,8 +862,16 @@ def _on_config_changed(self, _) -> None:
         if not self.upgrade.idle:
             logger.debug("Early exit on_config_changed: upgrade in progress")
             return
-        # update config on every run
-        self.update_config()
+
+        try:
+            # update config on every run
+            self.update_config()
+        except ValueError as e:
+            self.unit.status = BlockedStatus(f"Configuration Error: {str(e)}")
```

Comment:
```suggestion
            self.unit.status = BlockedStatus(f"Configuration Error. Please check the logs")
            logger.error("Invalid configuration: %s", str(e))
```
Looks like the validation message can be quite long and it's not going to be very readable in the status.

Better to dump it in the logs and inform the user to check that in the status message.